### PR TITLE
Consistency between L.circle and L.circleMarker

### PR DESCRIPTION
**Pull request by IvanSanchez**
Implements #3785

### CODE REVIEW

File: src/layer/vector/Circle.js [hunk 1]
```diff
@@ -62,6 +62,10 @@ L.Circle = L.CircleMarker.extend({
 	}
 });
 
-L.circle = function (latlng, radius, options) {
-	return new L.Circle(latlng, radius, options);
+L.circle = function (latlng, options, legacyOptions) {
+	if (typeof options === 'number') {
+		// Backwards compatibility with 0.7.x factory (latlng, radius, options?)
+		options = L.extend(L.extend({}, legacyOptions), {radius: options});
```
Comment (yohanboniface) [inline src/layer/vector/Circle.js]: When `options` is already an object, `radius` will be wrong, no?

Comment (yohanboniface) [inline src/layer/vector/Circle.js]: `if (typeof options === 'number')` sorry :)

**Comment by yohanboniface** [comment]
I think it deserves one unittest for new API, and one unittest for backward compat, what do you think?

**Comment by mourner** [comment]
Committed with https://github.com/Leaflet/Leaflet/commit/3924d49bcdd4d88099b0d9a3196ebcd29479a4f7

**Pull request closed without merging**